src/main/java/au/com/rayh/XCodeBuilder.java

```diff
@@ -716,47 +737,59 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
                 }
 
-                List<String> packageCommandLine = new ArrayList<String>();
-                packageCommandLine.add(getGlobalConfiguration().getXcrunPath());
-                packageCommandLine.add("-sdk");
-
-                if (!StringUtils.isEmpty(sdk)) {
-                    packageCommandLine.add(sdk);
-                } else {
-                    packageCommandLine.add(buildPlatform);
-                }
-                packageCommandLine.addAll(Lists.newArrayList("PackageApplication", "-v", app.absolutize().getRemote(), "-o", ipaLocation.absolutize().getRemote()));
-                if (!StringUtils.isEmpty(embeddedProfileFile)) {
-                    packageCommandLine.add("--embed");
-                    packageCommandLine.add(embeddedProfileFile);
-                }
-                if (!StringUtils.isEmpty(codeSigningIdentity) && signIpaOnXcrun) {
-                    packageCommandLine.add("--sign");
-                    packageCommandLine.add(codeSigningIdentity);
-                }
 
+                List<String> packageCommandLine = new ArrayList<String>();
+                packageCommandLine.add(getGlobalConfiguration().getXcodebuildPath());
+                packageCommandLine.addAll(Lists.newArrayList("-exportArchive", "-archivePath", archive.absolutize().getRemote(), "-exportPath", ipaOutputPath.absolutize().getRemote(), "-exportOptionsPlist", exportPlistLocation.absolutize().getRemote()));
                 returnCode = launcher.launch().envs(envs).stdout(listener).pwd(projectRoot).cmds(packageCommandLine).join();
                 if (returnCode > 0) {
                     listener.getLogger().println("Failed to build " + ipaLocation.absolutize().getRemote());
                     return false;
                 }
+                //rename exported ipa
+                FilePath exportedIpa = ipaOutputPath.child(archive.getBaseName() + ".ipa");
+                if (exportedIpa.exists()) {
+                    exportedIpa.renameTo(ipaLocation);
+                }
+
 
                 // also zip up the symbols, if present
-                FilePath dSYM = app.withSuffix(".dSYM");
-                if (dSYM.exists()) {
-                    returnCode = launcher.launch().envs(envs).stdout(listener).pwd(buildDirectory).cmds("ditto", "-c", "-k", "--keepParent", "-rsrc", dSYM.absolutize().getRemote(), ipaOutputPath.child(baseName + "-dSYM.zip").absolutize().getRemote()).join();
+                listener.getLogger().println("Archiving dSYM");
+                List<FilePath> dSYMs = buildDirectory.absolutize().child(configuration + "-" + buildPlatform).list(new DSymFileFilter());
+
+                if (dSYMs.isEmpty()) {
+                    listener.getLogger().println("No dSYM file found in " + buildDirectory.absolutize().child(configuration + "-" + buildPlatform) + "!");
+                }
+
+                for(FilePath dSYM : dSYMs) {
```

Comment:
also move this for loop into the above if check since if dSYMs is null we crash here with nullptr exception as well.